HW02_A_Graph-Fails.Rmd

```diff
@@ -51,36 +56,46 @@ ggplot(data = mpg) +
 ### Graph Fail 3
 Two mistakes in this graph. First, I wanted to make the the points slightly bolder, but changing the alpha to 2 does nothing. What does alpha do and what does setting it to 2 do? What could be done instead if I want the points slightly bigger? 
 
+ans: alpha controls transparency and ranges from 0 (invisible) to 1(solid); a value of 2 should result in an error. To make the size of the point bigger we should use the size parameter.
+
 Second, I wanted to move the legend on top of the graph since there aren't any points there, putting it at approximately the point/ordered pair (5, 40). How do you actually do this? Also, how do you remove the legend title ("class")? Finally, how would you remove the plot legend completely?
+
+ans:  legend.position = c(5,40) will not work since the coordinates range from (0,0)bottom-left to (1,1) top-right
+      theme(legend.title=element_blank()) removes legend title
+      theme(legend.position="none") removes legend plot completely
 ```{r}
 mpg %>% 
 ggplot() + 
-  geom_point(mapping = aes(x = displ, y = hwy, color = class), alpha = 2) + 
+  geom_point(mapping = aes(x = displ, y = hwy, color = class), alpha = 1, size=3) + 
   theme(legend.direction = "horizontal") + 
-  theme(legend.position = c(5, 40))
-
+  theme(legend.position = c(.6, .8)) 
 ```
 
 ### Graph Fail 4
 I wanted just one smoothing line. Just one line, to show the general relationship here. But that's not happening. Instead I'm getting 3 lines, why and fix it please?
 
+ans: aesthetic grouping default is based on number of discrete variables, so we need to override this by specifying group=1. (Q: Why 1 though? It seems to be a dummy variable?)

```

Comment:
Since any code in the ggplot layer applies to any following layer, you can just move the color specification to the geom_point layer instead. That way, the geom_smooth layer would only make one like from the x and y aesthetics specified in the first layer.